src/main/java/seedu/address/logic/parser/EditSocialMediaCommandParser.java

```diff
@@ -33,7 +34,12 @@ public EditSocialMediaCommand parse(String userInput) throws ParseException {
         }
 
         String newSocialMedia = argMultimap.getValue(PREFIX_SOCIAL_MEDIA).get();
-        Index oneBasedIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+        Index oneBasedIndex;
+        try {
+            oneBasedIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+        } catch (ParseException e) {
+            throw new ParseException(MESSAGE_INVALID_SOCIAL_MEDIA_INDEX);
+        }
```

Comment:
i think we should throw an invalid index parse exception here to tell the user that the index must be in range `1 to 2^31`? 

i.e. the original parseException

Reply:
I think there's no need for the user to know anything about 2^31. Any index out of bound should return "invalid index"